app/controllers/shared/api.php

```diff
@@ -37,41 +37,60 @@
     /*
      * Abuse Check
      */
-    $timeLimit = new TimeLimit($route->getLabel('abuse-key', 'url:{url},ip:{ip}'), $route->getLabel('abuse-limit', 0), $route->getLabel('abuse-time', 3600), $db);
-    $timeLimit->setNamespace('app_'.$project->getId());
-    $timeLimit
-        ->setParam('{userId}', $user->getId())
-        ->setParam('{userAgent}', $request->getUserAgent(''))
-        ->setParam('{ip}', $request->getIP())
-        ->setParam('{url}', $request->getHostname().$route->getPath())
-    ;
+    $abuseKeyLabel = $route->getLabel('abuse-key', 'url:{url},ip:{ip}');
+    $timeLimitArray = [];
+    if (is_array($abuseKeyLabel)) {
+        for ($i = 0; $i < count($abuseKeyLabel); $i++)  {
+            $timeLimit = new TimeLimit($abuseKeyLabel[$i], $route->getLabel('abuse-limit', 0), $route->getLabel('abuse-time', 3600), $db);
+            $timeLimit->setNamespace('app_'.$project->getId());
+            $timeLimit
+                ->setParam('{userId}', $user->getId())
+                ->setParam('{userAgent}', $request->getUserAgent(''))
+                ->setParam('{ip}', $request->getIP())
+                ->setParam('{url}', $request->getHostname().$route->getPath())
+            ;
+            $timeLimitArray[] = $timeLimit;
+        }
+    } else {
+        $timeLimit = new TimeLimit($abuseKeyLabel, $route->getLabel('abuse-limit', 0), $route->getLabel('abuse-time', 3600), $db);
+        $timeLimit->setNamespace('app_'.$project->getId());
+        $timeLimit
+            ->setParam('{userId}', $user->getId())
+            ->setParam('{userAgent}', $request->getUserAgent(''))
+            ->setParam('{ip}', $request->getIP())
+            ->setParam('{url}', $request->getHostname().$route->getPath())
+        ;
+        $timeLimitArray[] = $timeLimit;
```

Comment:
To be more dry, you can just check if `$abuseKeyLabel` is not an array and push it to `$timeLimitArray` - that way you will only have one occurence of `new TimeLimit()`.

Something like:
```php
if (!is_array($abuseKeyLabel)) {
    $timeLimitArray[] = $abuseKeyLabel;
}
```

That way you can iterate over the array everytime 👍🏻